modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationPolicyDeploymentTestCase.java

```diff
@@ -239,6 +243,103 @@ public void appliesMultipleApplicationPolicies() throws Exception {
     assertManualExecutionsCount(2);
   }
 
+  @Test
+  public void duplicatedApplicationPolicy() throws Exception {
+    policyManager.registerPolicyTemplate(fooPolicyFileBuilder.getArtifactFile());
+
+    ApplicationFileBuilder applicationFileBuilder = createExtensionApplicationWithServices(APP_WITH_EXTENSION_PLUGIN_CONFIG,
+                                                                                           helloExtensionV1Plugin);
+    addPackedAppFromBuilder(applicationFileBuilder);
+
+    startDeployment();
+    assertApplicationDeploymentSuccess(applicationDeploymentListener, applicationFileBuilder.getId());
+
+    policyManager.addPolicy(applicationFileBuilder.getId(), fooPolicyFileBuilder.getArtifactId(),
+                            new PolicyParametrization(FOO_POLICY_ID, pointparameters -> true, 1,
+                                                      singletonMap(POLICY_PROPERTY_KEY, POLICY_PROPERTY_VALUE),
+                                                      getResourceFile("/fooPolicy.xml"), emptyList()));
+
+    try {
+      policyManager.addPolicy(applicationFileBuilder.getId(), fooPolicyFileBuilder.getArtifactId(),
+                              new PolicyParametrization(FOO_POLICY_ID, pointparameters -> true, 2,
+                                                        singletonMap(POLICY_PROPERTY_KEY, POLICY_PROPERTY_VALUE),
+                                                        getResourceFile("/fooPolicy.xml"), emptyList()));
+      fail("Policy Re-Application should have failed.");
```

Comment:
Use the `ExpectedException` rule (`expectedEx ` already being used in this test class)